### PR TITLE
add spec.version and status.version to AzureASOManagedControlPlane

### PR DESCRIPTION
**Pull request by nojnhuh**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:

This PR adds the `spec.version` and `status.version` fields to AzureASOManagedControlPlane per the CAPI contract. `spec.version` has no functional impact yet, but that will be wired up soon. `status.version` should still show the actual running version of Kubernetes per AKS.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Part of #4713

**Special notes for your reviewer**:

- [ ] cherry-pick candidate

**TODOs**:

- [X] squashed commits
- [ ] includes documentation
- [X] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: exp/controllers/azureasomanagedcontrolplane_controller_test.go [hunk 1]
```diff
@@ -224,7 +224,8 @@ func TestAzureASOManagedControlPlaneReconcile(t *testing.T) {
 				Namespace: cluster.Namespace,
 			},
 			Status: asocontainerservicev1.ManagedCluster_STATUS{
-				Fqdn: ptr.To("endpoint"),
+				Fqdn:                     ptr.To("endpoint"),
+				CurrentKubernetesVersion: ptr.To("Current"),
```
Comment (willie-yao) [inline exp/controllers/azureasomanagedcontrolplane_controller_test.go:228]: nit: Changing "Current" to a string in semver format might make more sense so we don't have to check for "vCurrent".

Comment (nojnhuh) [reply exp/controllers/azureasomanagedcontrolplane_controller_test.go:228]: What exactly do you mean by "so we don't have to check for 'vCurrent'"?

Comment (jackfrancis) [reply exp/controllers/azureasomanagedcontrolplane_controller_test.go:228]: Looks like there is some "v" prefix injection here. If so should we have two tests, one where we validate the "v" injection (passing in a real version in the test case e.g., "1.30.0" makes sense IMO) and then one where we pass in a string that starts w/ "v" (e.g., "v1.30.0") and verify that there isn't double-injection (e.g., "vv1.30.0")

Comment (nojnhuh) [reply exp/controllers/azureasomanagedcontrolplane_controller_test.go:228]: AFAIK AKS never adds the "v" prefix, but should we account for that anyway?

Comment (willie-yao) [reply exp/controllers/azureasomanagedcontrolplane_controller_test.go:228]: Ah I was just being extra picky about the formatting of the string and passing in a real version. I don't think there's anything wrong with keeping it as "Current".

Also if AKS never adds it, I think it should be good as is.

**Pull request merged**